src/Hamiltonians/ExtendedHubbardReal1D.jl

```diff
@@ -24,21 +24,24 @@ Hamiltonian can be either real or complex, depending on the choice of `boundary_
     factor ``\\exp(iθ)`` for a hop to the right and ``\\exp(−iθ)`` for a hop to the left.
     With this choice the Hamiltonian will have a complex `eltype` whereas otherwise the
     `eltype` is determined by the type of the parameters `t`, `u`, and `v`.
+* `interaction`: the interaction type. The following values are supported:
+  * `:nearest_neighbour`: nearest neighbour interaction (default), i.e. ``f_{i-j} = δ_{i,j-1}``.
+  * `<:Number`: inverse distance interaction, i.e. ``f_{i-j} = (|i-j|)^{-interaction}``.
 
 See also [`HubbardRealSpace`](@ref).
 """
-struct ExtendedHubbardReal1D{TT,A<:SingleComponentFockAddress,U,V,T,BOUNDARY_CONDITION} <: AbstractHamiltonian{TT}
+struct ExtendedHubbardReal1D{TT,A<:SingleComponentFockAddress,U,V,T,BOUNDARY_CONDITION,INTERACTION} <: AbstractHamiltonian{TT}
     address::A
 end
 
 # addr for compatibility.
-function ExtendedHubbardReal1D(addr; u=1.0, v=1.0, t=1.0, boundary_condition = :periodic)
+function ExtendedHubbardReal1D(addr; u=1.0, v=1.0, t=1.0, boundary_condition=:periodic, interaction=:nearest_neighbour)
```

Comment:
```suggestion
function ExtendedHubbardReal1D(addr; u=1.0, v=1.0, t=1.0, boundary_condition=:periodic, interaction=:nearest_neighbor)
```
We use American spelling in the code.

Reply:
Changes have been made.